content-src/asrouter/templates/Trailhead/Trailhead.jsx

```diff
@@ -53,82 +37,18 @@ export class Trailhead extends React.PureComponent {
     return this.props.document.getElementById("trailheadDialog");
   }
 
-  async componentWillMount() {
-    FLUENT_FILES.forEach(file => {
-      const link = document.head.appendChild(document.createElement("link"));
-      link.href = file;
-      link.rel = "localization";
-    });
-
-    await this.componentWillUpdate(this.props);
-  }
-
-  // Get the fxa data if we don't have it yet from mount or update
-  async componentWillUpdate(props) {
-    if (props.fxaEndpoint && !this.fxaMetricsInitialized) {
-      try {
-        this.fxaMetricsInitialized = true;
-        const url = new URL(
-          `${
-            props.fxaEndpoint
-          }/metrics-flow?entrypoint=activity-stream-firstrun&form_type=email`
-        );
-        this.addUtmParams(url);
-        const response = await fetch(url, { credentials: "omit" });
-        if (response.status === 200) {
-          const { deviceId, flowId, flowBeginTime } = await response.json();
-          this.setState({ deviceId, flowId, flowBeginTime });
-        } else {
-          props.dispatch(
-            ac.OnlyToMain({
-              type: at.TELEMETRY_UNDESIRED_EVENT,
-              data: {
-                event: "FXA_METRICS_FETCH_ERROR",
-                value: response.status,
-              },
-            })
-          );
-        }
-      } catch (error) {
-        props.dispatch(
-          ac.OnlyToMain({
-            type: at.TELEMETRY_UNDESIRED_EVENT,
-            data: { event: "FXA_METRICS_ERROR" },
-          })
-        );
-      }
-    }
-  }
-
   componentDidMount() {
-    // We need to remove hide-main since we should show it underneath everything that has rendered
-    this.props.document.body.classList.remove("hide-main");
-
-    // Add inline-onboarding class to disable fixed search header and fixed positioned settings icon
-    this.props.document.body.classList.add("inline-onboarding");
-
-    // The rest of the page is "hidden" when the modal is open
-    if (this.props.message.content) {
-      this.props.document
-        .getElementById("root")
-        .setAttribute("aria-hidden", "true");
-
-      // Start with focus in the email input box
-      this.dialog.querySelector("input[name=email]").focus();
-    } else {
-      // No modal overlay, let the user scroll and deal them some cards.
-      this.props.document.body.classList.remove("welcome");
-
-      if (this.props.message.includeBundle || this.props.message.cards) {
-        this.revealCards();
-      }
+    // The rest of the page is "hidden" to screen readers when the modal is open
+    this.props.document
+      .getElementById("root")
+      .setAttribute("aria-hidden", "true");
```

Comment:
I guess we don't want to scope creep too much for this refactoring, so I guess a followup to set aria-hidden for the other FirstRun instead of only Trailhead. Although to be clear, this wasn't necessary before when showing "fxa_overlay" because delayed removing "hide-main".

Reply:
Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1567957